packages/react-devtools-shared/src/devtools/views/Settings/SettingsContext.js

```diff
@@ -68,6 +68,15 @@ type Context = {|
 const SettingsContext = createContext<Context>(((null: any): Context));
 SettingsContext.displayName = 'SettingsContext';
 
+function useLocalStorageWithLog<T>(
+  key: string,
+  initialValue: T | (() => T),
+): [T, (value: T | (() => T)) => void] {
+  return useLocalStorage<T>(key, initialValue, {
```

Comment:
nit: you probably can inline this:

```js
const [theme, setTheme] = useLocalStorage<Theme>('React::DevTools::theme', 'auto', SETTINGS_CHANGED_EVENT)

function useLocalStorage(key, initialValue, eventName?) {
   const loggerEvent = {
        event_name: eventName,
       ...
   }
}
```

Reply:
But what if we want to log something other than the event name?

Reply:
What's an example? I'm not really that opinionated. Also down to pass in a loggerEvent and then append to it :)